### DISCUSSION TITLE
Copy and Paste Single Node

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#5989] [Work In Progress - GUI]
Currently it is possible to copy and paste text within a node when editing.

Could we make it so that if you copy when a node is selected it's whole text is copied to the clipboard and then if we paste into the IDE with text in the clipboard it should be added as a new node? This wouldn't need to be clever and if the value was invalid the user would need to edit and correct, e.g. pasting a string value would need to be quoted by the user afterwards.

Ideally, we would be able to detect pasting a table and translate it to a serialized format so we can trivially create a look up table. Assuming we can do the above this would hopefully be something we could expand later.

CC: @farmaazon @sylwiabr

**Comment by wdanilo** [comment]
It is definitely possible. We have pretty nice clipboard manager implemented. Just telling its possible.

**Comment by farmaazon** [comment]
Implemented in https://github.com/enso-org/enso/issues/6261.
